yeelight/main.py

```diff
@@ -59,15 +61,25 @@ def _command(f, *args, **kw):
     if result:
         return result[0]
 
+def get_ip_address(ifname):
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    return socket.inet_ntoa(fcntl.ioctl(
+        s.fileno(),
+        0x8915,  # SIOCGIFADDR
+        struct.pack('256s', ifname[:15])
```

Comment:
This is IPv4-only, right? Is there any chance it'll break IPv6-based discovery?

Reply:
Looks like this is only used when an interface is specified, so it looks like it won't break existing functionality.

Reply:
Yep, IPv4 only. It actually fails on IPv6-only interfaces. Maybe we should consider using [netifaces](https://pypi.python.org/pypi/netifaces/) and implement a more robust check.
Also, it is not uncommon to have many IPv6 address on the same interface, maybe at that point it would be better to let the user specify the address to be used, rather than choosing a random one/the first one.

Reply:
Hmm, right. I think this is good enough for now though.